prime-router/src/main/kotlin/JurisdictionalFilters.kt

```diff
@@ -0,0 +1,95 @@
+     */
+    val name: String
+
+    /**
+     * @args values passed to the filter
+     *
+     * @return the Selection object to be used to filter the
+     * tablesaw table.
+     */
+    fun getSelection(args: List<String>, table: Table): Selection
+}
+
+/**
+ * Implements the most basic filter.
+ * matches(columnName, regex)
+ */
+class Matches : JurisdictionalFilter {
+    override val name = "matches"
+
+    override fun getSelection(args: List<String>, table: Table): Selection {
+        if (args.size != 2) error("Expecting two args to filter $name:  (columnName, regex)")
+        val columnName = args[0]
+        val pattern = args[1]
+        return table.stringColumn(columnName).matchesRegex(pattern)
+    }
+}
+
+/**
+ * This may or may not be a unicorn.
+ */
+class FilterByCounty : JurisdictionalFilter {
+    override val name = "filterByCounty"
+
+    // @todo need tons of error checking.
+    override fun getSelection(args: List<String>, table: Table): Selection {
+        if (args.size != 2) error("Expecting two args to filter $name:  (TwoLetterState, County)")
+        val patientState = table.stringColumn("standard.patient_state")
+            ?: error("Unable to filterByCounty:  column standard.patient_state not found.")
+        val patientCounty = table.stringColumn("standard.patient_county")
+            ?: error("Unable to filterByCounty:  column standard.patient_county not found.")
+        val facilityState = table.stringColumn("standard.ordering_facility_state")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_state not found.")
+        val facilityCounty = table.stringColumn("standard.ordering_facility_county")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_county not found.")
+
+        // Try to be very loose on county matching.   Anything with the county name embedded is ok.
+        val countyRegex = "(?i).*${args[1]}.*"
+        // Overall, this is "true" if either the patient is in the county/state
+        //   OR, if the facility is in the county/state.
+        val patientSelection = patientState.isEqualTo(args[0]).and(patientCounty.matchesRegex(countyRegex))
+        val facilitySelection = facilityState.isEqualTo(args[0]).and(facilityCounty.matchesRegex(countyRegex))
+        return patientSelection.or(facilitySelection)
+    }
+}
+
+object JurisdictionalFilters {
+    /**
+     * filterFunction must be of form "funName(arg1, arg2, etc)"
+     */
+    fun parseJurisdictionalFilter(filterFunction: String): Pair<String, List<String>> {
+// REMOVE        val match = Regex("([a-zA-Z0-9]+)\\x28([a-z, \\x2E_\\x2DA-Z0-9]*)\\x29").find(filterFunction)
+        // Using a permissive match in the (arg1, arg2) section, to allow most regexs to be passed as args.
+        // Somehow this works with "(?i).*Pima.*", I guess because the \\x29 matches rightmost ')' char
+        val match = Regex("([a-zA-Z0-9]+)\\x28(.*)\\x29").find(filterFunction)
```

Comment:
The generic (.*) in the middle was necessitated by the need to pass gloppy regex'es in.